docs/pgo.md

```diff
@@ -0,0 +1,39 @@
+# [Profile-Guided Optimization (PGO)](https://en.wikipedia.org/wiki/Profile-guided_optimization)
+
+PGO is a technique to automatically improve a program's runtime performance based on profiling.
+
+PGO is applied to the final executable, so it cannot be applied for this Rust library for anyone to use. If you want to benefit from PGO for maximum performance, you will need to follow the steps below.
```

Comment:
Actually, PGO can be applied to a library, if anyone decides to compile the library as a static library. However, it's pretty uncommon for Rust but anyway - it's possible.

I can suggest replacing it with a safe phrase "PGO should be applied to the final executable. If you want to benefit from PGO for maximum performance, you will need to follow the steps below."